app/controllers/katello/api/v2/capsule_content_controller.rb

```diff
@@ -51,7 +51,7 @@ def remove_lifecycle_environment
     param :environment_id, Integer, :desc => N_('Id of the environment to limit the synchronization on')
     def sync
       find_environment if params[:environment_id]
-      task = async_task(::Actions::Katello::CapsuleContent::Sync, capsule_content, :environment => @environment)
+      task = async_task(::Actions::Katello::CapsuleContent::Sync, capsule_content.capsule, :environment_id => @environment.try(:id))
```

Comment:
Why the change to environment_id? It seems like it's just used to find the environment below (same question goes for the other ids too)